javaslang/src/test/java/javaslang/collection/VectorPropertyTest.java

```diff
@@ -0,0 +1,260 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import javaslang.Function2;
+import javaslang.Tuple;
+import javaslang.Tuple2;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.function.Function;
+
+import static javaslang.Function2.constant;
+import static javaslang.collection.BitMappedTrie.BRANCHING_BASE;
+import static javaslang.collection.BitMappedTrie.branchingFactor;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class VectorPropertyTest {
+    @Before
+    public void setUp() { BRANCHING_BASE = 2; }
```

Comment:
`gradle` would run them in forked JVMs and all tests should run with `BRANCHING_BASE = 2` also, but you're right that this introduces unwanted statefulness (and the fix is easy)
Thanks!.